src/platform/nxp/zephyr/ota/OTAImageProcessorImpl.cpp

```diff
@@ -96,10 +97,16 @@ CHIP_ERROR OTAImageProcessorImpl::Apply()
 #ifdef CONFIG_CHIP_OTA_REQUESTOR_REBOOT_ON_APPLY
     if (!err)
     {
+        PlatformMgr().HandleServerShuttingDown();
+        /*
+         * Restart the device in order to apply the update image.
+         * This should be done with a delay so the device has enough time to send
+         * the state-transition event when applying the update.
+         */
+        ChipLogProgress(SoftwareUpdate, "Restarting device, will reboot in %d seconds ...", delayBeforeReboot);
         return SystemLayer().StartTimer(
-            System::Clock::Milliseconds32(CHIP_DEVICE_CONFIG_OTA_REQUESTOR_REBOOT_DELAY_MS),
+            System::Clock::Milliseconds32(delayBeforeReboot * 1000 + MIN_REBOOT_DELAY),
```

Comment:
what is the unit of measurement for delayBeforeReboot? At first I thought MS, but the clock is already Milliseconds32 ... is this thousentofams ? that seems odd.

Reply:
Usage of MIN_REBOOT_DELAY is also suspect - it seems to be a delta, otherwise I would expect some std::min call not a plus.

Reply:
@andy31415, The unit for `delayBeforeReboot` is seconds. I have added the unit to the variable name.
MIN_REBOOT_DELAY is used here as a delta, I have updated the name to `deltaRebootDelayMs`.
Sorry for the confusion.